src/tests/Microsoft.PowerFx.Core.Tests.Shared/UserDefinedTypeTests.cs

```diff
@@ -93,6 +93,15 @@ public void TestUserDefinedType(string typeDefinition, string expectedDefinedTyp
 
         // Have named formulas and udf in the script
         [InlineData("NAlias := Type(Number);X := 5; ADDX(n:Number): Number = n + X; SomeType := Type(UntypedObject)", 2)]
+
+        // Have RecordOf with/ without errors
```

Comment:
>with/ without errors [](http://example.com/codeflow?start=25&length=20)

Curious: what are the invalid cases here? From a quick glance they all look valid :-)

Reply:
`T1 := Type(RecordOf([Number]));` this is invalid that why 3 valid types instead of 4 , we do not want to support inlining within `RecordOf` as it is just reduntant